lib/ansible/modules/cloud/amazon/aws_acm.py

```diff
@@ -0,0 +1,356 @@
+    privateKey: "{{ lookup('file', 'key.pem' ) }}"
+    name_tag: my_cert # to be applied through an AWS tag as  "Name":"my_cert"
+    region: ap-southeast-2 # AWS region
+
+- name: create/update a certificate with a chain
+  aws_acm:
+    certificate: "{{ lookup('file', 'cert.pem' ) }}"
+    privateKey: "{{ lookup('file', 'key.pem' ) }}"
+    name_tag: my_cert
+    certificate_chain: "{{ lookup('file', 'chain.pem' ) }}"
+    state: present
+    region: ap-southeast-2
+  register: cert_create
+
+- name: print ARN of cert we just created
+  debug:
+    var: cert_create.certificate.arn
+
+- name: delete the cert we just created
+  aws_acm:
+    name_tag: my_cert
+    state: absent
+    region: ap-southeast-2
+
+- name: delete a certificate with a particular ARN
+  aws_acm:
+    certificate_arn: "arn:aws:acm:ap-southeast-2:123456789012:certificate/01234567-abcd-abcd-abcd-012345678901"
+    state: absent
+    region: ap-southeast-2
+
+- name: delete all certificates with a particular domain name
+  aws_acm:
+    domain_name: acm.ansible.com
+    state: absent
+    region: ap-southeast-2
+
+'''
+
+RETURN = '''
+certificate:
+  description: Information about the certificate which was uploaded
+  type: complex
+  returned: when C(state) is C(present)
+  contains:
+    arn:
+      description: The ARN of the certificate in ACM
+      type: str
+      returned: when C(state) is C(present)
+      sample: "arn:aws:acm:ap-southeast-2:123456789012:certificate/01234567-abcd-abcd-abcd-012345678901"
+    domain_name:
+      description: The domain name encoded within the public certificate
+      type: str
+      returned: when C(state) is C(present)
+      sample: acm.ansible.com
+arns:
+  description: A list of the ARNs of the certificates in ACM which were deleted
+  type: list
+  returned: when C(state) is C(absent)
+  sample:
+   - "arn:aws:acm:ap-southeast-2:123456789012:certificate/01234567-abcd-abcd-abcd-012345678901"
+'''
+
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.aws.acm import ACMServiceManager
+
+
+# For converting PEM bodies to something that can be easily compared
+# e.g. ignore case, strip trailing whitespace
+#
+# There's probably a standard crypto library to do this, I'm writing 
+# another function here to minimize dependencies.
+# The lack of rigour we get by not importing a proper crypto lib
+# should result in only errors where equal certs are considered different,
+# in which case the only impact is that changed=True when it should be changed=False.
+# The other kind of error (different certs being considered equal) should not happen.
+def standardize_pem(pem):
+    if pem is None:
+        return('')
+    # eliminate whitespace
+    for c in [' ', '\n', '\t']:
+        pem = pem.replace(c, '')
+
+    # I'm not sure whether the number of dashes at the top and bottom can vary
```

Comment:
Valid PEM files can have 5 or 6 dashes. Depending on what kind of PEM format you get. Collapsing them to one is definitely a good idea.

Reply:
I've replaced my custom compare function with a module_utils one.